components/CalendarPopup.js

```diff
@@ -1,38 +1,52 @@
 import styled from "styled-components";
 import Link from "next/link";
+import BulbIcon from "assets/calendarIcons/bulb.svg";
+import { useRouter } from "next/router";
+import { format } from "date-fns";
 
 export default function CalendarPopup({
   getEmotionsForDay,
   selectedDay,
   onClosePopup,
 }) {
+  const formattedDate = format(selectedDay.date, "yyyy-MM-dd");
+  const router = useRouter();
+  const emotionsForDay = getEmotionsForDay(selectedDay.date);
```

Comment:
if you pass down selectedDay and getEmotionForDay, you could find the `emotionsForDay` outside of this component in the parent and pass them down instead. This would reduce the props by one and simplify the data flow.